src/content/actions/receive-profile.js

```diff
@@ -232,24 +234,91 @@ export function receiveProfileFromWeb(profile: Profile): ThunkAction {
   };
 }
 
-export function errorReceivingProfileFromWeb(error: any): Action {
+export function temporaryErrorReceivingProfileFromWeb(error: TemporaryError): Action {
   return {
-    type: 'ERROR_RECEIVING_PROFILE_FROM_WEB',
+    type: 'TEMPORARY_ERROR_RECEIVING_PROFILE_FROM_WEB',
     error,
   };
 }
 
+export function fatalErrorReceivingProfileFromWeb(error: Error): Action {
+  return {
+    type: 'FATAL_ERROR_RECEIVING_PROFILE_FROM_WEB',
+    error,
+  };
+}
+
+function _wait(delayMs) {
+  return new Promise(resolve => setTimeout(resolve, delayMs));
+}
+
+type FetchProfileArgs = {
+  url: string,
+  onTemporaryError: TemporaryError => void,
+};
+/**
+ * Tries to fetch a profile on URL url. If the profile is not found,
+ * onTemporaryError is called with an appropriate error, we wait 1 second, and
+ * then tries again. If we still can't find the profile after 11 tries, we throw
+ * a permanent error.
+ * @param {string} url URL to fetch
```

Comment:
We can see why jsdocs are bad; in this case it is properly obsolete, as I wrote it when I only had a `url` parameter and forgot to update it :)

I'd like to at least document when onTemporaryError is called and when the return promise is resolved/rejected. But maybe we don't need a JSdoc for this ?